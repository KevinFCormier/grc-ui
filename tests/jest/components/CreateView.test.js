/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2019. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
'use strict'

import React from 'react'
import CreationView from '../../../src-web/components/CreationView'
import renderer from 'react-test-renderer'

const existing =
{
  'clusterLabels':[
    {
      'key':'cloud',
      'value':'IBM'
    },
    {
      'key':'datacenter',
      'value':'toronto'
    },
    {
      'key':'environment',
      'value':'Dev'
    },
    {
      'key':'owner',
      'value':'marketing'
    },
    {
      'key':'region',
      'value':'US'
    },
    {
      'key':'vendor',
      'value':'ICP'
    }
  ],
  'compliances':[
    {
      'name':'grc-policy',
      'namespace':'mcm',
      'annotations':{
        'policy.mcm.ibm.com/categories':'',
        'policy.mcm.ibm.com/controls':'',
        'policy.mcm.ibm.com/standards':'',
        'seed-generation':'1'
      },
      'spec':{
        'complianceType':'musthave',
        'namespaces':{
          'exclude':[
            'kube-*'
          ],
          'include':[
            'default'
          ]
        },
        'object-templates':[
          {
            'complianceType':'musthave',
            'objectDefinition':{
              'apiVersion':'policy/v1beta1',
              'kind':'PodSecurityPolicy',
              'metadata':{
                'annotations':{
                  'seccomp.security.alpha.kubernetes.io/allowedProfileNames':'*'
                },
                'name':'restricted-mcm'
              },
              'spec':{
                'allowPrivilegeEscalation':false,
                'allowedCapabilities':[
                  '*'
                ],
                'fsGroup':{
                  'rule':'RunAsAny'
                },
                'hostIPC':false,
                'hostNetwork':true,
                'hostPID':false,
                'hostPorts':[
                  {
                    'max':65535,
                    'min':1000
                  }
                ],
                'privileged':false,
                'runAsUser':{
                  'rule':'RunAsAny'
                },
                'seLinux':{
                  'rule':'RunAsAny'
                },
                'supplementalGroups':{
                  'rule':'RunAsAny'
                },
                'volumes':[
                  '*'
                ]
              }
            },
            'status':{
              'Validity':{

              }
            }
          }
        ],
        'remediationAction':'enforce'
      },
      '__typename':'ExistingCompliance'
    },
    {
      'name':'grc-policy2',
      'namespace':'mcm',
      'annotations':{
        'policy.mcm.ibm.com/categories':'',
        'policy.mcm.ibm.com/controls':'',
        'policy.mcm.ibm.com/standards':'',
        'seed-generation':'1'
      },
      'spec':{
        'complianceType':'musthave',
        'namespaces':{
          'exclude':[
            'kube-*'
          ],
          'include':[
            'default'
          ]
        },
        'object-templates':[
          {
            'complianceType':'musthave',
            'objectDefinition':{
              'apiVersion':'policy/v1beta1',
              'kind':'PodSecurityPolicy',
              'metadata':{
                'annotations':{
                  'seccomp.security.alpha.kubernetes.io/allowedProfileNames':'*'
                },
                'name':'restricted-mcm'
              },
              'spec':{
                'allowPrivilegeEscalation':false,
                'allowedCapabilities':[
                  '*'
                ],
                'fsGroup':{
                  'rule':'RunAsAny'
                },
                'hostIPC':false,
                'hostNetwork':true,
                'hostPID':false,
                'hostPorts':[
                  {
                    'max':65535,
                    'min':1000
                  }
                ],
                'privileged':false,
                'runAsUser':{
                  'rule':'RunAsAny'
                },
                'seLinux':{
                  'rule':'RunAsAny'
                },
                'supplementalGroups':{
                  'rule':'RunAsAny'
                },
                'volumes':[
                  '*'
                ]
              }
            },
            'status':{
              'Validity':{

              }
            }
          }
        ],
        'remediationAction':'enforce'
      },
      '__typename':'ExistingCompliance'
    },
    {
      'name':'policy-iam',
      'namespace':'mcm',
      'annotations':{
        'kubectl.kubernetes.io/last-applied-configuration':'{"apiVersion":"policy.mcm.ibm.com/v1alpha1","kind":"Policy","metadata":{"annotations":{"policy.mcm.ibm.com/categories":"SystemAndInformationIntegrity","policy.mcm.ibm.com/controls":"MutationAdvisor","policy.mcm.ibm.com/standards":"NIST"},"name":"policy-iam","namespace":"mcm"},"spec":{"policy-templates":[{"objectDefinition":{"apiVersion":"mcm-grcpolicy.ibm.com/v1alpha1","kind":"IamPolicy","metadata":{"label":{"category":"System-Integrity"},"name":"iam-policy-example"},"spec":{"maxClusterRoleBindingUsers":5,"namespaceSelector":{"exclude":["kube-system"],"include":["default","kube-*"]},"remediationAction":"enforce"}}}]}}\n',
        'policy.mcm.ibm.com/categories':'SystemAndInformationIntegrity',
        'policy.mcm.ibm.com/controls':'MutationAdvisor',
        'policy.mcm.ibm.com/standards':'NIST',
        'seed-generation':'1'
      },
      'spec':{
        'namespaces':{

        },
        'policy-templates':[
          {
            'objectDefinition':{
              'apiVersion':'mcm-grcpolicy.ibm.com/v1alpha1',
              'kind':'IamPolicy',
              'metadata':{
                'label':{
                  'category':'System-Integrity'
                },
                'name':'iam-policy-example'
              },
              'spec':{
                'maxClusterRoleBindingUsers':5,
                'namespaceSelector':{
                  'exclude':[
                    'kube-system'
                  ],
                  'include':[
                    'default',
                    'kube-*'
                  ]
                },
                'remediationAction':'enforce'
              }
            },
            'status':{
              'Validity':{

              }
            }
          }
        ]
      },
      '__typename':'ExistingCompliance'
    },
    {
      'name':'policy-ma',
      'namespace':'mcm',
      'annotations':{
        'policy.mcm.ibm.com/categories':'SystemAndInformationIntegrity',
        'policy.mcm.ibm.com/controls':'MutationAdvisor',
        'policy.mcm.ibm.com/standards':'NIST',
        'seed-generation':'1'
      },
      'spec':{
        'namespaces':{

        },
        'policy-templates':[
          {
            'objectDefinition':{
              'apiVersion':'policies.ibm.com/v1alpha1',
              'kind':'MutationPolicy',
              'metadata':{
                'label':{
                  'category':'System-Integrity'
                },
                'name':'mutation-policy-example'
              },
              'spec':{
                'conditions':{
                  'ownership':[
                    'ReplicaSet',
                    'Deployment',
                    'DeamonSet',
                    'ReplicationController'
                  ]
                },
                'namespaceSelector':{
                  'exclude':[
                    'kube-system'
                  ],
                  'include':[
                    'default',
                    'kube-*'
                  ]
                },
                'remediationAction':'enforce'
              }
            },
            'status':{
              'Validity':{

              }
            }
          }
        ],
        'remediationAction':'enforce'
      },
      '__typename':'ExistingCompliance'
    },
    {
      'name':'policy-pod',
      'namespace':'mcm',
      'annotations':{
        'policy.mcm.ibm.com/categories':'SystemAndCommunicationsProtections,SystemAndInformationIntegrity',
        'policy.mcm.ibm.com/controls':'MutationAdvisor,VA',
        'policy.mcm.ibm.com/standards':'NIST,HIPAA',
        'seed-generation':'1'
      },
      'spec':{
        'complianceType':'musthave',
        'namespaces':{
          'exclude':[
            'kube*'
          ],
          'include':[
            'default'
          ]
        },
        'object-templates':[
          {
            'complianceType':'musthave',
            'objectDefinition':{
              'apiVersion':'v1',
              'kind':'Pod',
              'metadata':{
                'name':'nginx-pod'
              },
              'spec':{
                'containers':[
                  {
                    'image':'nginx:1.7.9',
                    'name':'nginx',
                    'ports':[
                      {
                        'containerPort':80
                      }
                    ]
                  }
                ]
              }
            },
            'status':{
              'Validity':{

              }
            }
          }
        ],
        'remediationAction':'inform'
      },
      '__typename':'ExistingCompliance'
    },
    {
      'name':'policy-role',
      'namespace':'mcm',
      'annotations':{
        'policy.mcm.ibm.com/categories':'SystemAndCommunicationsProtections,SystemAndInformationIntegrity',
        'policy.mcm.ibm.com/controls':'MutationAdvisor,VA',
        'policy.mcm.ibm.com/standards':'NIST,HIPAA',
        'seed-generation':'1'
      },
      'spec':{
        'namespaces':{
          'exclude':[
            'kube*'
          ],
          'include':[
            'default'
          ]
        },
        'remediationAction':'inform',
        'role-templates':[
          {
            'apiVersion':'roletemplate.mcm.ibm.com/v1alpha1',
            'complianceType':'musthave',
            'metadata':{
              'creationTimestamp':null,
              'name':'operator-role-policy'
            },
            'rules':[
              {
                'complianceType':'musthave',
                'policyRule':{
                  'apiGroups':[
                    'extensions',
                    'apps'
                  ],
                  'resources':[
                    'deployments'
                  ],
                  'verbs':[
                    'get',
                    'list',
                    'watch',
                    'create',
                    'delete',
                    'patch'
                  ]
                }
              }
            ],
            'selector':{
              'matchLabels':{
                'dev':'true'
              }
            },
            'status':{
              'Validity':{

              }
            }
          }
        ]
      },
      '__typename':'ExistingCompliance'
    },
    {
      'name':'policy1',
      'namespace':'mcm',
      'annotations':{
        'policy.mcm.ibm.com/categories':'SystemAndInformationIntegrity, RBAC',
        'policy.mcm.ibm.com/controls':'MutationAdvisor',
        'policy.mcm.ibm.com/standards':'NIST',
        'seed-generation':'2'
      },
      'spec':{
        'complianceType':'musthave',
        'namespaces':{
          'exclude':[
            'kube*'
          ],
          'include':[
            'default'
          ]
        },
        'object-templates':[
          {
            'complianceType':'musthave',
            'objectDefinition':{
              'apiVersion':'rbac.authorization.k8s.io/v1',
              'kind':'RoleBinding',
              'metadata':{
                'name':'operate-pods-rolebinding',
                'namespace':'default'
              },
              'roleRef':{
                'apiGroup':'rbac.authorization.k8s.io',
                'kind':'Role',
                'name':'operator'
              },
              'subjects':[
                {
                  'apiGroup':'rbac.authorization.k8s.io',
                  'kind':'User',
                  'name':'admin'
                }
              ]
            },
            'status':{
              'Validity':{

              }
            }
          },
          {
            'complianceType':'musthave',
            'objectDefinition':{
              'apiVersion':'policy/v1beta1',
              'kind':'PodSecurityPolicy',
              'metadata':{
                'annotations':{
                  'seccomp.security.alpha.kubernetes.io/allowedProfileNames':'*'
                },
                'name':'restricted-mcm'
              },
              'spec':{
                'allowPrivilegeEscalation':false,
                'allowedCapabilities':[
                  '*'
                ],
                'fsGroup':{
                  'rule':'RunAsAny'
                },
                'hostIPC':false,
                'hostNetwork':true,
                'hostPID':false,
                'hostPorts':[
                  {
                    'max':65535,
                    'min':1000
                  }
                ],
                'privileged':false,
                'runAsUser':{
                  'rule':'RunAsAny'
                },
                'seLinux':{
                  'rule':'RunAsAny'
                },
                'supplementalGroups':{
                  'rule':'RunAsAny'
                },
                'volumes':[
                  '*'
                ]
              }
            },
            'status':{
              'Validity':{

              }
            }
          },
          {
            'complianceType':'musthave',
            'objectDefinition':{
              'apiVersion':'networking.k8s.io/v1',
              'kind':'NetworkPolicy',
              'metadata':{
                'name':'deny-from-other-namespaces',
                'namespace':'default'
              },
              'spec':{
                'ingress':[
                  {
                    'from':[
                      {
                        'podSelector':{

                        }
                      }
                    ]
                  }
                ],
                'podSelector':{
                  'matchLabels':null
                }
              }
            },
            'status':{
              'Validity':{

              }
            }
          },
          {
            'complianceType':'musthave',
            'objectDefinition':{
              'apiVersion':'v1',
              'kind':'LimitRange',
              'metadata':{
                'name':'mem-limit-range'
              },
              'spec':{
                'limits':[
                  {
                    'default':{
                      'memory':'512Mi'
                    },
                    'defaultRequest':{
                      'memory':'256Mi'
                    },
                    'type':'Container'
                  }
                ]
              }
            },
            'status':{
              'Validity':{

              }
            }
          }
        ],
        'remediationAction':'inform',
        'role-templates':[
          {
            'apiVersion':'roletemplate.mcm.ibm.com/v1alpha1',
            'complianceType':'musthave',
            'metadata':{
              'creationTimestamp':null,
              'name':'operator-role'
            },
            'rules':[
              {
                'complianceType':'mustnothave',
                'policyRule':{
                  'apiGroups':[
                    'core'
                  ],
                  'resources':[
                    'secrets'
                  ],
                  'verbs':[
                    'get',
                    'list',
                    'watch',
                    'delete',
                    'create',
                    'update',
                    'patch'
                  ]
                }
              },
              {
                'complianceType':'musthave',
                'policyRule':{
                  'apiGroups':[
                    'core'
                  ],
                  'resources':[
                    'pods'
                  ],
                  'verbs':[
                    'get',
                    'list',
                    'watch'
                  ]
                }
              }
            ],
            'selector':{
              'matchLabels':{
                'dev':'true'
              }
            },
            'status':{
              'Validity':{

              }
            }
          }
        ]
      },
      '__typename':'ExistingCompliance'
    }
  ],
  '__typename':'Existing'
}



describe('CreationView component', () => {
  const setGetPolicyJSON = jest.fn()
  it('renders as expected', () => {
    const component = renderer.create(
      <CreationView
        existing={existing}
        setGetPolicyJSON={setGetPolicyJSON}
      />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})
