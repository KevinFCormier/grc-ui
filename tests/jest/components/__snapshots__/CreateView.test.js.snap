// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreationView component renders as expected 1`] = `
<div
  className="creation-view"
>
  <div
    className="creation-view-controls-container"
  >
    <div
      className="creation-view-controls"
    >
      <div
        className="creation-view-controls-textbox"
      >
        <div
          className="creation-view-controls-textbox-title"
        >
          Name
        </div>
        <div
          className="bx--form-item"
        >
          <input
            className="bx--text-input bx--text__input"
            disabled={false}
            id="policy-name"
            onChange={[Function]}
            onClick={[Function]}
            placeholder={undefined}
            type="text"
            value="policy-grc"
          />
        </div>
      </div>
      <div
        className="creation-view-controls-multiselect"
      >
        <div
          className="creation-view-controls-multiselect-title"
        >
          Specifications
          <div
            className="bx--tooltip-icon"
          >
            <button
              aria-label="The parameter section that describes the rules to validate a cluster."
              className="bx--tooltip__trigger bx--tooltip--icon__top"
            >
              <svg
                className="info-icon"
              >
                <use
                  href="#diagramIcons_info"
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          aria-invalid={undefined}
          aria-label="Choose an item"
          className="bx--multi-select bx--combo-box bx--list-box"
          data-invalid={undefined}
          onClick={[Function]}
          onKeyDown={[Function]}
          role="listbox"
          tabIndex="-1"
        >
          <div
            aria-expanded={false}
            aria-haspopup={true}
            aria-label="open menu"
            className="bx--list-box__field"
            data-toggle={true}
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex="0"
            type="button"
          >
            <input
              aria-activedescendant={null}
              aria-autocomplete="list"
              aria-expanded={false}
              autoComplete="off"
              className="bx--text-input"
              disabled={false}
              id="downshift-0-input"
              onBlur={[Function]}
              onChange={[Function]}
              onKeyDown={[Function]}
              placeholder="Select how to validate clusters"
              role="combobox"
              value=""
            />
            
            <div
              className="bx--list-box__menu-icon"
            >
              <svg
                alt="Open menu"
                aria-label="Open menu"
                className={undefined}
                fill={undefined}
                fillRule="evenodd"
                focusable="false"
                height="5"
                name={undefined}
                role="img"
                style={undefined}
                viewBox="0 0 10 5"
                width="10"
              >
                <title>
                  Open menu
                </title>
                <path
                  d="M0 0l5 4.998L10 0z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        className="creation-view-controls-multiselect"
      >
        <div
          className="creation-view-controls-multiselect-title"
        >
          Cluster selector
          <div
            className="bx--tooltip-icon"
          >
            <button
              aria-label="The parameter field where you select which cluster the policy is applied to. When you enter a value, a placement policy and placement binding policy are created. Note: A PlacementPolicy and PlacementBinding resource is required to apply your policy to a cluster."
              className="bx--tooltip__trigger bx--tooltip--icon__top"
            >
              <svg
                className="info-icon"
              >
                <use
                  href="#diagramIcons_info"
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          aria-invalid={undefined}
          aria-label="Choose an item"
          className="bx--multi-select bx--combo-box bx--list-box"
          data-invalid={undefined}
          onClick={[Function]}
          onKeyDown={[Function]}
          role="listbox"
          tabIndex="-1"
        >
          <div
            aria-expanded={false}
            aria-haspopup={true}
            aria-label="open menu"
            className="bx--list-box__field"
            data-toggle={true}
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex="0"
            type="button"
          >
            <input
              aria-activedescendant={null}
              aria-autocomplete="list"
              aria-expanded={false}
              autoComplete="off"
              className="bx--text-input"
              disabled={false}
              id="downshift-1-input"
              onBlur={[Function]}
              onChange={[Function]}
              onKeyDown={[Function]}
              placeholder="Select what clusters to validate"
              role="combobox"
              value=""
            />
            
            <div
              className="bx--list-box__menu-icon"
            >
              <svg
                alt="Open menu"
                aria-label="Open menu"
                className={undefined}
                fill={undefined}
                fillRule="evenodd"
                focusable="false"
                height="5"
                name={undefined}
                role="img"
                style={undefined}
                viewBox="0 0 10 5"
                width="10"
              >
                <title>
                  Open menu
                </title>
                <path
                  d="M0 0l5 4.998L10 0z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        className="creation-view-controls-checkbox"
      >
        <div
          className="bx--form-item bx--checkbox-wrapper"
        >
          <input
            checked={true}
            className="bx--checkbox"
            id="policy-enforce"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="bx--checkbox-label checkbox"
            htmlFor="policy-enforce"
            title={null}
          >
            <span
              className="bx--visually-hidden"
            >
              
            </span>
          </label>
        </div>
        <div>
          Enforce
        </div>
        <div
          className="bx--tooltip-icon"
        >
          <button
            aria-label="The remediation action for your policy. Select Enforce to automatically remediate your cluster based on your policies. When Enforce is not selected, the the remediation action is inform."
            className="bx--tooltip__trigger bx--tooltip--icon__top"
          >
            <svg
              className="info-icon"
            >
              <use
                href="#diagramIcons_info"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        className="creation-view-controls-multiselect"
      >
        <div
          className="creation-view-controls-multiselect-title"
        >
          Standards
          <div
            className="bx--tooltip-icon"
          >
            <button
              aria-label="The name or names of security standards the policy is trying to validate."
              className="bx--tooltip__trigger bx--tooltip--icon__top"
            >
              <svg
                className="info-icon"
              >
                <use
                  href="#diagramIcons_info"
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          aria-invalid={undefined}
          aria-label="Choose an item"
          className="bx--multi-select bx--combo-box bx--list-box"
          data-invalid={undefined}
          onClick={[Function]}
          onKeyDown={[Function]}
          role="listbox"
          tabIndex="-1"
        >
          <div
            aria-expanded={false}
            aria-haspopup={true}
            aria-label="open menu"
            className="bx--list-box__field"
            data-toggle={true}
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex="0"
            type="button"
          >
            <input
              aria-activedescendant={null}
              aria-autocomplete="list"
              aria-expanded={false}
              autoComplete="off"
              className="bx--text-input"
              disabled={false}
              id="downshift-2-input"
              onBlur={[Function]}
              onChange={[Function]}
              onKeyDown={[Function]}
              placeholder="Select standards this policy belongs to"
              role="combobox"
              value=""
            />
            
            <div
              className="bx--list-box__menu-icon"
            >
              <svg
                alt="Open menu"
                aria-label="Open menu"
                className={undefined}
                fill={undefined}
                fillRule="evenodd"
                focusable="false"
                height="5"
                name={undefined}
                role="img"
                style={undefined}
                viewBox="0 0 10 5"
                width="10"
              >
                <title>
                  Open menu
                </title>
                <path
                  d="M0 0l5 4.998L10 0z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        className="creation-view-controls-multiselect"
      >
        <div
          className="creation-view-controls-multiselect-title"
        >
          Categories
          <div
            className="bx--tooltip-icon"
          >
            <button
              aria-label="The security control category that the policy applies to."
              className="bx--tooltip__trigger bx--tooltip--icon__top"
            >
              <svg
                className="info-icon"
              >
                <use
                  href="#diagramIcons_info"
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          aria-invalid={undefined}
          aria-label="Choose an item"
          className="bx--multi-select bx--combo-box bx--list-box"
          data-invalid={undefined}
          onClick={[Function]}
          onKeyDown={[Function]}
          role="listbox"
          tabIndex="-1"
        >
          <div
            aria-expanded={false}
            aria-haspopup={true}
            aria-label="open menu"
            className="bx--list-box__field"
            data-toggle={true}
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex="0"
            type="button"
          >
            <input
              aria-activedescendant={null}
              aria-autocomplete="list"
              aria-expanded={false}
              autoComplete="off"
              className="bx--text-input"
              disabled={false}
              id="downshift-3-input"
              onBlur={[Function]}
              onChange={[Function]}
              onKeyDown={[Function]}
              placeholder="Select categories this policy belongs to"
              role="combobox"
              value=""
            />
            
            <div
              className="bx--list-box__menu-icon"
            >
              <svg
                alt="Open menu"
                aria-label="Open menu"
                className={undefined}
                fill={undefined}
                fillRule="evenodd"
                focusable="false"
                height="5"
                name={undefined}
                role="img"
                style={undefined}
                viewBox="0 0 10 5"
                width="10"
              >
                <title>
                  Open menu
                </title>
                <path
                  d="M0 0l5 4.998L10 0z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        className="creation-view-controls-multiselect"
      >
        <div
          className="creation-view-controls-multiselect-title"
        >
          Controls
          <div
            className="bx--tooltip-icon"
          >
            <button
              aria-label="The name of the security control that is being checked."
              className="bx--tooltip__trigger bx--tooltip--icon__top"
            >
              <svg
                className="info-icon"
              >
                <use
                  href="#diagramIcons_info"
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          aria-invalid={undefined}
          aria-label="Choose an item"
          className="bx--multi-select bx--combo-box bx--list-box"
          data-invalid={undefined}
          onClick={[Function]}
          onKeyDown={[Function]}
          role="listbox"
          tabIndex="-1"
        >
          <div
            aria-expanded={false}
            aria-haspopup={true}
            aria-label="open menu"
            className="bx--list-box__field"
            data-toggle={true}
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex="0"
            type="button"
          >
            <input
              aria-activedescendant={null}
              aria-autocomplete="list"
              aria-expanded={false}
              autoComplete="off"
              className="bx--text-input"
              disabled={false}
              id="downshift-4-input"
              onBlur={[Function]}
              onChange={[Function]}
              onKeyDown={[Function]}
              placeholder="Select controls this policy belongs to"
              role="combobox"
              value=""
            />
            
            <div
              className="bx--list-box__menu-icon"
            >
              <svg
                alt="Open menu"
                aria-label="Open menu"
                className={undefined}
                fill={undefined}
                fillRule="evenodd"
                focusable="false"
                height="5"
                name={undefined}
                role="img"
                style={undefined}
                viewBox="0 0 10 5"
                width="10"
              >
                <title>
                  Open menu
                </title>
                <path
                  d="M0 0l5 4.998L10 0z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="creation-view-yaml"
  >
    <div
      className="creation-view-yaml-header"
    >
      <div
        className="creation-view-yaml-header-title"
      >
        Policy YAML
      </div>
      <div
        aria-label={undefined}
        className="creation-view-yaml-header-toolbar"
        id={undefined}
        role="region"
      >
        <div
          className="editor-bar"
        >
          <div
            aria-label="Search"
            className="editor-bar-search"
            id="Search"
            role="region"
          >
            <div
              aria-labelledby="search-name-label"
              className="bx--search bx--search--sm"
              role="search"
            >
              <svg
                alt=""
                aria-label=""
                className="bx--search-magnifier"
                fill={undefined}
                fillRule="evenodd"
                height="16"
                name={undefined}
                role="img"
                style={undefined}
                viewBox="0 0 16 16"
                width="16"
              >
                <title>
                  
                </title>
                <path
                  d="M6.5 12a5.5 5.5 0 1 0 0-11 5.5 5.5 0 0 0 0 11zm4.936-1.27l4.563 4.557-.707.708-4.563-4.558a6.5 6.5 0 1 1 .707-.707z"
                />
              </svg>
              <label
                className="bx--label"
                htmlFor="search-name"
                id="search-name-label"
              >
                
              </label>
              <input
                aria-label="Search"
                className="bx--search-input"
                id="search-name"
                onChange={[Function]}
                placeholder="Search"
                type="text"
              />
              <button
                aria-label={undefined}
                className="bx--search-close bx--search-close--hidden"
                onClick={[Function]}
                type="button"
              >
                <svg
                  alt="Provide a description that will be used as the title"
                  aria-label="Provide a description that will be used as the title"
                  className={undefined}
                  fill={undefined}
                  fillRule="evenodd"
                  height="16"
                  name={undefined}
                  role="img"
                  style={undefined}
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <title>
                    Provide a description that will be used as the title
                  </title>
                  <path
                    d="M8 6.586L5.879 4.464 4.464 5.88 6.586 8l-2.122 2.121 1.415 1.415L8 9.414l2.121 2.122 1.415-1.415L9.414 8l2.122-2.121-1.415-1.415L8 6.586zM8 16A8 8 0 1 1 8 0a8 8 0 0 1 0 16z"
                  />
                </svg>
              </button>
            </div>
          </div>
          <div
            aria-label={undefined}
            className="editor-bar-button  disabled"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex="0"
            title={undefined}
          >
            <svg
              height="16px"
              width="16px"
            >
              <use
                href="#diagramIcons_previous"
              />
            </svg>
          </div>
          <div
            aria-label={undefined}
            className="editor-bar-button  disabled"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex="0"
            title={undefined}
          >
            <svg
              height="16px"
              width="16px"
            >
              <use
                href="#diagramIcons_next"
              />
            </svg>
          </div>
          <div
            className="editor-bar-spacer"
          />
          <div
            aria-label={undefined}
            className="editor-bar-button  disabled"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex="0"
            title={undefined}
          >
            <svg
              height="16px"
              width="16px"
            >
              <use
                href="#diagramIcons_undo"
              />
            </svg>
          </div>
          <div
            aria-label={undefined}
            className="editor-bar-button  disabled"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex="0"
            title={undefined}
          >
            <svg
              height="16px"
              width="16px"
            >
              <use
                href="#diagramIcons_redo"
              />
            </svg>
          </div>
          <div
            aria-label={undefined}
            className="editor-bar-button  disabled"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex="0"
            title={undefined}
          >
            <svg
              height="16px"
              width="16px"
            >
              <use
                href="#diagramIcons_restore"
              />
            </svg>
          </div>
          <div
            className="editor-bar-spacer"
          />
        </div>
      </div>
    </div>
    
    <div
      className="yamlEditorContainer"
    >
      <div
        id="brace-editor"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      />
    </div>
  </div>
</div>
`;
